src/plugins/matrixops/unique.cpp

```diff
@@ -0,0 +1,253 @@
+        {
+        case node_data_type_bool:
+            return unique0d(extract_boolean_value_strict(
+                std::move(args[0]), name_, codename_));
+
+        case node_data_type_int64:
+            return unique0d(extract_integer_value_strict(
+                std::move(args[0]), name_, codename_));
+
+        case node_data_type_double:
+            return unique0d(extract_numeric_value_strict(
+                std::move(args[0]), name_, codename_));
+
+        case node_data_type_unknown:
+            return unique0d(
+                extract_numeric_value(std::move(args[0]), name_, codename_));
+
+        default:
+            break;
+        }
+
+        HPX_THROW_EXCEPTION(hpx::bad_parameter,
+            "phylanx::execution_tree::primitives::unique::unique0d",
+            generate_error_message(
+                "the unique primitive requires for all arguments to "
+                "be numeric data types"));
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    template <typename T>
+    primitive_argument_type unique::unique1d(ir::node_data<T> && arg) const
+    {
+        blaze::DynamicVector<T> a = arg.vector();
+        // Sorting the vector
+        std::sort(a.begin(), a.end());
+
+        // Use std::unique to remove duplicacy
+        auto ip = std::unique(a.begin(), a.end());
+
+        // Resizing the vector to remove undifined elements
+        a.resize(std::distance(a.begin(), ip));
+
+        return primitive_argument_type{std::move(a)};
+    }
+
+    primitive_argument_type unique::unique1d(primitive_arguments_type && args)
+        const
+    {
+        switch (extract_common_type(args[0]))
+        {
+        case node_data_type_bool:
+            return unique1d(extract_boolean_value_strict(
+                std::move(args[0]), name_, codename_));
+
+        case node_data_type_int64:
+            return unique1d(extract_integer_value_strict(
+                std::move(args[0]), name_, codename_));
+
+        case node_data_type_double:
+            return unique1d(extract_numeric_value_strict(
+                std::move(args[0]), name_, codename_));
+
+        case node_data_type_unknown:
+            return unique1d(
+                extract_numeric_value(std::move(args[0]), name_, codename_));
+
+        default:
+            break;
+        }
+
+        HPX_THROW_EXCEPTION(hpx::bad_parameter,
+            "phylanx::execution_tree::primitives::unique::unique1d",
+            generate_error_message(
+                "the arange primitive requires for all arguments to "
+                "be numeric data types"));
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    template <typename T>
+    primitive_argument_type unique::unique2d(ir::node_data<T> && arg) const
+    {
+        auto data = arg.matrix();
+        auto d = data.data();
+
+        std::sort(d, d + data.rows() * data.spacing());
```

Comment:
This might get you wrong results as the matrix rows could be padded.

Reply:
@hkaiser Since it's padded with zeros I first sort the padded vector and then remove the first data.rows()*(data.spacing()-data.columns()) elements which were added by padding.

Reply:
@scheherzade how do you know that the zeros are at the beginning after sorting?

Reply:
@hkaiser Yes they are, I checked it with some examples.